esp/esp/program/modules/handlers/studentregphasezero.py

```diff
@@ -0,0 +1,220 @@
+                context['form'] = form
+                return render_to_response('program/modules/studentregphasezero/submit.html', request, context)
+
+            else:
+                context['lottery_group'] = PhaseZeroRecord.objects.filter(user=user, program=prog)[0]
+                context['lottery_size'] = len(context['lottery_group'].user.all())
+                return render_to_response('program/modules/studentregphasezero/confirmation.html', request, context)
+
+        elif Permission.user_has_perm(user, 'Student/Classes/PhaseZero', program=prog):
+            if in_lottery:
+                context['lottery_group'] = PhaseZeroRecord.objects.filter(user=user, program=prog)[0]
+                context['lottery_size'] = len(context['lottery_group'].user.all())
+                return render_to_response('program/modules/studentregphasezero/confirmation.html', request, context)
+            else:
+                if request.method == 'POST':
+                    form = SubmitForm(request.POST, program=prog)
+                    if form.is_valid():
+                        form.save(user, prog)
+                    context['lottery_group'] = PhaseZeroRecord.objects.filter(user=user, program=prog)[0]
+                    context['lottery_size'] = len(context['lottery_group'].user.all())
+                    self.send_confirmation_email(user)
+                    return render_to_response('program/modules/studentregphasezero/confirmation.html', request, context)
+                else:
+                    form = SubmitForm(program=prog)
+                    context['form'] = form
+                    return render_to_response('program/modules/studentregphasezero/submit.html', request, context)
+        else:
+            if in_lottery:
+                if Tag.getBooleanTag('student_lottery_run', prog, default=False):
+                    #Sorry page
+                    return render_to_response('program/modules/studentregphasezero/sorry.html', request, context)
+                else:
+                    #Lottery has not yet been run page
+                    return render_to_response('program/modules/studentregphasezero/notyet.html', request, context)
+            else:
+                #Generic error page
+                return render_to_response('errors/program/phasezero_closed.html', request, context)
+
+    @aux_call
+    @needs_student
+    def studentlookup(self, request, tl, one, two, module, extra, prog, newclass = None):
+
+        # Search for students with names that start with search string
+        if not 'username' in request.GET or 'username' in request.POST:
+            return self.goToCore(tl)
+
+        return StudentRegPhaseZero.studentlookup_logic(request, tl, one, two, module, extra, prog, newclass)
+
+    @staticmethod
+    def studentlookup_logic(request, tl, one, two, module, extra, prog, newclass = None):
+        limit = 10
+        from esp.web.views.json_utils import JsonResponse
```

Comment:
import should go at the top of the file unless there's a reason not to.  (This is a bit faster since it means the import won't happen in a request, and is generally the convention to make it easy to find all the imports.)